lib/typescript/model/Conversation.ts

```diff
@@ -22,3 +27,34 @@ export interface Conversation {
   createdAt: Date;
   lastMessage: Message;
 }
+
+export function conversationNotes(conversation: Conversation) {
+  return Object.keys(conversation.metadata.notes || {})
+    .map(noteId => {
+      return {
+        id: noteId,
+        text: conversation.metadata.notes[noteId]['text'],
+        timestamp: conversation.metadata.notes[noteId]['timestamp']
+          ? new Date(parseInt(conversation.metadata.notes[noteId]['timestamp']))
```

Comment:
So here the timestamp can either be a string or a `Date`, but in the typescript interface it says `string`. We should decide for one and stick with it 🙏

Reply:
But the date comes from the backend as a string to prevent metadata discrepancies, so it has to be defined as such in the metadata interface as well. However, we want to work with the note's date as an object in the functional code, so in the `Note` interface (`lib/typescript/model/Note`) it is defined as a Date. Does that make sense or should we move the date conversion to all places where it's needed? This seems counterproductive given the sort operations. I hope this makes sense 🤞

Reply:
No, it makes perfect sense. I only wanted to point out that in the ternary you return a `: ''` as the default when the field does not exist. Thereby you break the `NoteModel` interface. So the default should also be some sort of date.

Suggestion regarding the millis as a string coming from the backend: [Here](https://github.com/airyhq/airy/blob/d53c4c0bc13c125a4f84bbd2f8f447f32fa4790e/backend/api/communication/src/main/java/co/airy/core/api/communication/ConversationsController.java#L211) we could store the String already as an ISO string using our [`isoFromMillis`](https://github.com/airyhq/airy/blob/0f3ab3554fbc41b0fe1f44d04fb32c858a1c84b3/lib/java/date/src/main/java/co/airy/date/format/DateFormat.java#L12) function. What do you think?

Reply:
Ah I see what you mean, I will set it to `null` to prevent rendering.

Concerning the `isoFromMillis`, the MetadataRepository and Metadata builder [do not accept](https://github.com/airyhq/airy/blob/d53c4c0bc13c125a4f84bbd2f8f447f32fa4790e/backend/model/metadata/src/main/java/co/airy/model/metadata/MetadataRepository.java#L14) anything other than a string so this would lead to a lot of refactoring. It could be done but is maybe not necessary?

Reply:
> Ah I see what you mean, I will set it to null to prevent rendering.
Good solution 👍 
>Concerning the isoFromMillis, the MetadataRepository and Metadata builder do not accept anything other than a string so this would lead to a lot of refactoring. It could be done but is maybe not necessary?
Yes and `isoFromMillis` returns an iso-compliant date string.

Reply:
> > Ah I see what you mean, I will set it to null to prevent rendering.
> > Good solution 👍
> > Concerning the isoFromMillis, the MetadataRepository and Metadata builder do not accept anything other than a string so this would lead to a lot of refactoring. It could be done but is maybe not necessary?
> > Yes and `isoFromMillis` returns an iso-compliant date string.

Oh oops I didn't realize. Fixed it now!